073-extend-feature-gates-to-all-operators.md

```diff
@@ -0,0 +1,114 @@
+The feature gates do not extent to the other operators (User and Topic Operators).
+
+## Motivation
+
+Although the feature gates currently don't apply to the other operators, some of them still affect them:
+* For example, the `UnidirectionalTopicOperator` feature gate controlled the mode in which the Topic Operator was running.
+  But it was using different environment variables to configure the right mode Topic Operator and not the feature gates configuration.
+* In the future, the `UseServerSideApply` feature gate (see [Proposal no. 52](https://github.com/strimzi/proposals/blob/main/052-k8s-server-side-apply.md) for more details) will apply to at least the User Operator.
+
+Additional feature gates that apply to the other operators might be added in the future.
+To have a unified way how to manage such feature gates, this proposal provides the plan for extending the feature gates to the User and Topic Operators as well.
+
+## Proposal
+
+The feature gates and the `STRIMZI_FEATURE_GATES` environment variable will keep its current format.
+The support for the `STRIMZI_FEATURE_GATES` environment variable will be added to the User and Topic Operators.
+The same feature gates will be defined and handled by every operator.
+However, if the operator doesn't have any use for the feature gate, it can simply ignore it in its code.
+
+When the User and Topic Operators are deployed through the Cluster Operator as part of the Kafka cluster, the user will configure the feature gates as today using the `STRIMZI_FEATURE_GATES` environment variable.
+And the Cluster Operator will then automatically pass the environment variable to the User and Topic Operators when deploying them.
+When the User and Topic Operators are deployed as standalone, user can configure the `STRIMZI_FEATURE_GATES` environment variable directly in their deployments.
+
+### Trade-offs and limitations
+
+This design has several trade-offs and limitations: 
+* When a feature gate is enabled or disabled in the Cluster Operator, it will cause rolling update of all the User and Topic Operators managed by it.
+  This will happen even if they in reality don't use given feature gate in their code and simply ignore it.
+  That said, changing the feature gates configuration is not something what is expected to be done very often.
+* The centrally managed feature gate configuration will also apply to all Kafka clusters managed by given Cluster Operator.
+  It cannot be applied to individual Kafka clusters which can be seen as a limitation.
+  But it does not differ significantly from the current state.
+
+### Advantages
+
+It has also aspects that can be considered as advantages:
+* The configuration of the feature gates is unchanged compared to how it works today.
+  It also maintains the simplicity of the configuration.
+  The user needs to set only one environment variable in the Cluster Operator to configure all feature gates.
```

Comment:
unless using standalone operators?

Reply:
Yes, correct. I added it there. But when using the standalone operators, you have completely standalone installations, so you cannot really expect it to be done somehow centrally.